presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -846,11 +843,12 @@ private boolean addNewExchangesAndBuffers()
     }
 
     @VisibleForTesting
+    @SuppressWarnings("NakedNotify")
```

Comment:
Why this warning? The `this.notifyAll()` call is inside a `synchronized (this)` block.